protocol/ntpshm/ntpshm.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright (c) Facebook, Inc. and its affiliates.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ntpshm
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+)
+
+// SHMKEY is a key of the first SHM segment
+const SHMKEY = 0x4e545030
+
+// IPC_CREAT create if key is nonexistent
+const IPC_CREAT = 00001000
+
+// IPC_EXCL fail if key exists
+const IPC_EXCL = 00002000
+
+/* NTPSHM Declaration of the SHM segment from ntp (ntpd/refclock_shm.c) */
+type NTPSHM struct {
+	mode                 int32
+	count                int32
+	clockTimeStampSec    uint
```

Comment:
In ntpd sources it has type of `time_t`, which is a **signed** integer on POSIX platform. Are you sure it's `uint`, not `int`?

that's also weird to leave those values platform-specific while others are strictly 32bit. Either everything should have strict size, or everything platform-specific, depending on how it's written.